### PR TITLE
Pinoreplace formatlogs#8

### PR DESCRIPTION
**Pull request by brianmugweru**
Replace bunyan with pino
Format logs as they are on google stack driver
Add generate request id function to module
Update testing

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,11 +23,15 @@
     "bunyan": "1.8.10",
     "dirty-chai": "1.2.2",
     "express": "4.15.2",
+    "ip": "1.1.5",
     "joi": "10.4.1",
+    "lodash": "4.17.11",
     "morgan": "1.8.1",
```
Comment (obengwilliam) [inline package.json]: Are we still using morgan and morgan json

File: package.json [hunk 2]
```diff
@@ -23,11 +23,15 @@
     "bunyan": "1.8.10",
     "dirty-chai": "1.2.2",
     "express": "4.15.2",
+    "ip": "1.1.5",
     "joi": "10.4.1",
+    "lodash": "4.17.11",
```
Comment (obengwilliam) [inline package.json]: dont use lodash, we want to keep the dependency small

File: package.json [hunk 3]
```diff
@@ -23,11 +23,15 @@
     "bunyan": "1.8.10",
     "dirty-chai": "1.2.2",
```
Comment (obengwilliam) [inline package.json]: push non dev packages to dependency and the other to devdepencies

File: package.json [hunk 4]
```diff
@@ -23,11 +23,15 @@
...
+    "ip": "1.1.5",
     "joi": "10.4.1",
+    "lodash": "4.17.11",
     "morgan": "1.8.1",
-    "morgan-json": "1.1.0"
+    "morgan-json": "1.1.0",
+    "pino": "5.11.1"
   },
   "devDependencies": {
+    "@kudobuzz/kbscripts": "0.0.5",
```
Comment (obengwilliam) [inline package.json]: Kindly add the linter

File: package.json [hunk 5]
```diff
@@ -36,6 +40,7 @@
     "expressjs": "1.0.1",
     "husky": "0.13.3",
     "mocha": "3.2.0",
+    "node-mocks-http": "1.7.3",
```
Comment (obengwilliam) [inline package.json]: replace node-mocks-http with nock

Comment (brianmugweru) [reply package.json]: node-mocks-http allows mocking of request object on its own, nock mocks a server which feels like integration testing

Comment (obengwilliam) [reply package.json]: they do the same thing, nock is used to mock a request

File: example/app.js [hunk 6]
```diff
@@ -1,4 +1,5 @@
-const log = require('../logger')()
+'use strict'
+const log = require('../logger')({ name: 'new', routeName: 'new route' })
```
Comment (obengwilliam) [inline example/app.js]: why do we need a routeName as part of the options

File: example/app.js [hunk 7]
```diff
@@ -9,16 +10,16 @@ const assignReqId = (req, res, next) => {
 
 app.use(assignReqId) // assignRequestId to each incoming request
```
Comment (obengwilliam) [inline example/app.js]: can u remove this we dont need to continue using this

File: package.json (hunk 5 above)
Comment (brianmugweru) [reply package.json]: nock mocks the http request, node-mocks-http mocks the request object passed to the middleware during a http request, so the two can not be used in place of each other... Been through nock's documentation and code for a while and they do not export any function to generate Request or Response Object

**Pull request merged**